packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -258,12 +258,23 @@ export class DebugConfigurationManager {
         });
     }
     protected async doCreate(model: DebugConfigurationModel): Promise<URI> {
-        const uri = new URI(model.workspaceFolderUri).resolve('.theia/launch.json');
+        await this.preferences.set('launch', {}); // create dummy launch.json in the correct place
+        const { configUri } = this.preferences.resolve('launch'); // get uri to write content to it
+        let uri: URI;
+        if (configUri && configUri.path.base === 'launch.json') {
+            uri = configUri;
+        } else { // fallback
+            uri = new URI(model.workspaceFolderUri).resolve('.theia/launch.json');
```

Comment:
we should use the first config path here, not `.theia`, i mean from `preferenceconfigurations.parths`

Reply:
fixed